kyaml/openapi/openapi.go

```diff
@@ -344,7 +344,36 @@ func (rs *ResourceSchema) Field(field string) *ResourceSchema {
 	return &ResourceSchema{Schema: &s}
 }
 
-// PatchStrategyAndKey returns the patch strategy and merge key extensions
+// PatchStrategyAndKeyList returns the patch strategy and complete merge key list
+func (rs *ResourceSchema) PatchStrategyAndKeyList() (string, []string) {
+	ps, found := rs.Schema.Extensions[kubernetesPatchStrategyExtensionKey]
+	if !found {
+		// empty patch strategy
+		return "", []string{}
+	}
+
+	mkList, found := rs.Schema.Extensions[kubernetesMergeKeyMapList]
+	if found {
+		//mkList is []interface, convert to []string
+		mkListStr := make([]string, len(mkList.([]interface{})))
+
+		for i, v := range mkList.([]interface{}) {
+			mkListStr[i] = fmt.Sprint(v)
```

Comment:
Interesting that you're using `fmt.Sprint`. Will that convert non-strings into odd-looking strings?

Reply:
I didn't know how to convert interfaces to strings so I looked it up and this was the first thing I found. In hindsight there are better ways so I've changed this